tensorflow/python/training/slot_creator.py

```diff
@@ -76,11 +76,16 @@ def _create_slot_var(primary, val, scope, validate_shape, shape, dtype):
     # remove "'linear//weight' + '/'" and ':0'.
     real_slot_name = slot.name[len(primary.op.name + "/"):-2]
     slice_info = primary._save_slice_info
-    slot._set_save_slice_info(variables.Variable.SaveSliceInfo(
-        slice_info.full_name + "/" + real_slot_name,
-        slice_info.full_shape[:],
-        slice_info.var_offset[:],
-        slice_info.var_shape[:]))
+    # supprot slot's shape not same as primary's shape
+    # example: primary's shape = [10, 20, 30], slot's shape =
+    # [], [10], [10, 20] or [10, 20, 30] is allowed
+    n = slot.shape.ndims
+    if (n > 0):
+      slot._set_save_slice_info(variables.Variable.SaveSliceInfo(
+          slice_info.full_name + "/" + real_slot_name,
+          slice_info.full_shape[:n],
```

Comment:
Isn't the full shape always just slot.shape? Why slice?

Reply:
A slot is a `Variable` created with the same first m-dimension as a primary `Tensor`. A slot is always scoped in the namespace of the primary object and	variable or `Tensor`
So, slot's slice info can be slice of the primary's slice info

Reply:
Yes, but the full shape is just "slot.shape" always. There's no need to slice.

Reply:
the 'slice_info.full_shape' is the primary's slice_info, not the slot's slice_info.
slot's slice_info may be different from primary's slice_info.

Reply:
Right, `slice_info.full_shape` would need to be sliced. But `slot.shape` wouldn't. This is nitpicky, so I'm fine leaving it slicing. But do you agree that `slot.shape == slice_info.full_shape[:n]`? If not, a unit test where it fails would help me understand.